src/main/java/com/netgrif/application/engine/petrinet/service/ProcessRoleService.java

```diff
@@ -21,61 +20,58 @@
 import org.bson.types.ObjectId;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Lazy;
 import org.springframework.stereotype.Service;
 
 import java.util.*;
 import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 @Service
 public class ProcessRoleService implements IProcessRoleService {
 
     private static final Logger log = LoggerFactory.getLogger(ProcessRoleService.class);
 
-    @Autowired
-    private IUserService userService;
-
-    @Autowired
-    private ProcessRoleRepository processRoleRepository;
-
-    @Autowired
-    private PetriNetRepository netRepository;
-
-    @Autowired
-    private UserDetailsServiceImpl userDetailsService;
-
-    @Autowired
-    private ApplicationEventPublisher publisher;
-
-    @Autowired
-    private RoleActionsRunner roleActionsRunner;
-
-    @Autowired
-    private IPetriNetService petriNetService;
-
-    @Autowired
-    private ISecurityContextService securityContextService;
+    private final IUserService userService;
+    private final ProcessRoleRepository processRoleRepository;
+    private final PetriNetRepository netRepository;
+    private final ApplicationEventPublisher publisher;
+    private final RoleActionsRunner roleActionsRunner;
+    private final IPetriNetService petriNetService;
+    private final ISecurityContextService securityContextService;
 
     private ProcessRole defaultRole;
-
     private ProcessRole anonymousRole;
 
+    public ProcessRoleService(IUserService userService, ProcessRoleRepository processRoleRepository,
+                              PetriNetRepository netRepository,
+                              ApplicationEventPublisher publisher, RoleActionsRunner roleActionsRunner,
+                              @Lazy IPetriNetService petriNetService, ISecurityContextService securityContextService) {
+        this.userService = userService;
+        this.processRoleRepository = processRoleRepository;
+        this.netRepository = netRepository;
+        this.publisher = publisher;
+        this.roleActionsRunner = roleActionsRunner;
+        this.petriNetService = petriNetService;
+        this.securityContextService = securityContextService;
+    }
+
     @Override
     public List<ProcessRole> saveAll(Iterable<ProcessRole> entities) {
         return processRoleRepository.saveAll(entities);
     }
 
     @Override
     public Set<ProcessRole> findByIds(Set<String> ids) {
-        return processRoleRepository.findAllBy_idIn(ids);
+        return StreamSupport.stream(processRoleRepository.findAllById(ids).spliterator(), false).collect(Collectors.toSet());
```

Comment:
The mentioned method `findAllById` is called from `CrudRepository` which is the superclass of `ProcessRoleRepository`. Previously was the method unnecessary overridden.

Reply:
The override changed the return type from `Iterable` to `Set`. Now we have to do this conversion in our code. Why is it being done with a serial stream? Why can't we just pass the returned `Iterable` into the constructor of some `Set` implementation?

Reply:
For backwards compatibility return type of the method should not change. The stream method is the quickest way to convert `Iterable` to `Set` because `Set` implementations don't have a constructor that accepts `Iterable` (at least none I know of).

Reply:
my bad. I missunderstood the documentation. Set implementations have a constructor that accepts a `Collection` and I thought, that since a `Collections` is also an `Iterrable`, then an `Iterrable` could be passed as an argument, but that is not the case.